### PR TITLE
authentication for login added

### PR DESCRIPTION
**Pull request by atykwonderland**
# Description

added the oauth feature for login 
- this is following the oauth with jwt tokens tutorial from fastapi
- does not include anything other than the actual login

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

i didn't know how to write a test for this but it can be tested through postman
- if anyone knows how to write a test for auth lmk and i will add one

- [x] through postman: 
1. run the api test to make sure u have a user saved in the db to login with then run the server
2. use postman: http://127.0.0.1:8000/token
3. go into the authorization tab
4. type: OAuth 2.0, Add authorization data to: Request Headers
5. click on: get new access token
6. grant type: password credentials, URL: same as above, username: in, password: wonderland
7. request token
8. if auth is successful, API shows 200 response

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I will rebase before merging

### CODE REVIEW

File: backend/app/main/crud.py [hunk 1]
```diff
@@ -147,6 +154,36 @@ def remove_user(session: Session, user_id: int):
...
+        return False
+    if not verify_password(password, user.password):
+        return False
+    return user
+
+
+# to get a string like this run:
+# openssl rand -hex 32
+SECRET_KEY = "0c4e63c7dc5c9aa52b419a1b1637af973744898204dd18ebcc0632a78ef051d0"
+ALGORITHM = "HS256"
```
Comment (Ty-Won) [inline backend/app/main/crud.py:173]: Hey alice great work, we shouldnt' include any secret keys in the PR. Please look into creating a .env file that is something hidden from git. Something similar is written in here: https://fastapi.tiangolo.com/advanced/settings/#read-env-vars-in-python except they just export the variable through the command line (You could do, but I suggest creating a file). To create an environment file, you would create a hidden .env file and put it in the gitignore. Once done, your code will have to target that file to extract variables in there. Check this out: https://fastapi.tiangolo.com/advanced/settings/#reading-a-env-file

**Comment by Ty-Won** [comment]
@atykwonderland  did you still need git help? this looks okay to me atm since it doesnt appear that you have unwanted commits

File: backend/app/main/crud.py (hunk 1 above)
Comment (atykwonderland) [reply backend/app/main/crud.py]: fixed with the .env file! pls check

File: backend/app/main/schemas.py [hunk 2]
```diff
@@ -98,3 +106,19 @@ class User(UserBase):
...
         orm_mode = True
+
+
+class TokenBase(BaseModel):
+    access_token: str
+    token_type: str
+
+
+class TokenCreate(TokenBase):
+    pass
```
Comment (Ty-Won) [inline backend/app/main/schemas.py:117]: Awesome job @atykwonderland! Do we need this class btw?

**Comment by Ty-Won** [comment]
@atykwonderland it looks good and def has my approval on its own! At this moment there are merge conflicts so if you can resolve them, I'll happily review them again :)

File: backend/app/main/schemas.py (hunk 2 above)
Comment (atykwonderland) [reply backend/app/main/schemas.py:117]: oops forgot to take tht out

**Pull request merged**